src/types.jl

```diff
@@ -539,8 +555,22 @@ end
 
 unflatten(t) = t
 
-function TermInterface.maketerm(::Type{<:BasicSymbolic}, head, args, type, metadata)
-    basicsymbolic(head, args, type, metadata)
+function TermInterface.maketerm(T::Type{<:BasicSymbolic}, head, args, metadata)
+    st = symtype(T)
+    pst = _promote_symtype(head, args)
+    # Use promoted symtype only if not a subtype of the existing symtype of T.
+    # This is useful when calling `maketerm(BasicSymbolic{Number}, (==), [true, false])` 
+    # Where the result would have a symtype of Bool. 
+    # Please see discussion in https://github.com/JuliaSymbolics/SymbolicUtils.jl/pull/609 
+    # TODO this should be optimized.
+    new_st = if pst === Bool 
+        pst 
+    elseif pst === Any || (st === Number && pst <: st) 
+        st
+    else 
+        pst 
+    end 
```

Comment:
This looks good! To ensure we're fully testing the new cases, could we add tests to reach 100% branch coverage here?

Reply:
Do you have any idea for a test structure? Also #565

Reply:
Since codecov is currently disabled, we probably have to do a thorough manual review of the code and tests to ensure all branches are adequately covered.
![IMG_7475](https://github.com/user-attachments/assets/8daec403-90e5-4b50-85d2-73f5364375d5)